prov/tcp/src/tcpx_progress.c

```diff
@@ -666,6 +685,24 @@ void tcpx_progress_rx(struct tcpx_ep *ep)
 		tcpx_ep_disable(ep, 0);
 }
 
+void tcpx_progress_err(struct tcpx_ep *ep)
+{
+	struct tcpx_xfer_entry *xfer;
+	uint32_t done;
+
+	done = ofi_bsock_async_done(&tcpx_prov, &ep->bsock);
+	while (!slist_empty(&ep->async_queue)) {
+		xfer = container_of(ep->async_queue.head,
+				    struct tcpx_xfer_entry, entry);
+		if (ofi_val32_gt(xfer->async_index, done))
```

Comment:
That's probably a stupid question but why not calling "xfer->async_index - done > 0" directly? Especially as both variables are uint32_t already.

Reply:
Not a stupid question at all.  These are 32-bit values that will wrap fairly quickly (probably seconds in a bw test).  If you look at the macro, it casts the results from u32 to int32, which handles the wrapping.  That works as long as index and done never drift farther apart than half the range.